karma.conf.js

```diff
@@ -2,39 +2,45 @@
 // https://karma-runner.github.io/0.13/config/configuration-file.html
 
 module.exports = function (config) {
-  config.set({
-    basePath: '',
-    frameworks: ['jasmine', '@angular-devkit/build-angular'],
-    plugins: [
-      require('karma-jasmine'),
-      require('karma-chrome-launcher'),
-      require('karma-remap-istanbul'),
-      require('@angular-devkit/build-angular/plugins/karma')
-    ],
-    files: [
+    config.set({
+        basePath: '',
+        frameworks: ['jasmine', '@angular-devkit/build-angular'],
+        plugins: [
+            require('karma-jasmine'),
+            require('karma-chrome-launcher'),
+            require('karma-jasmine-html-reporter'),
+            require('karma-coverage-istanbul-reporter'),
+            require('@angular-devkit/build-angular/plugins/karma')
+        ],
+        client: {
+            clearContext: false // leave Jasmine Spec Runner output visible in browser
+        },
+        mime: {
+            'text/x-typescript': ['ts', 'tsx']
+        },
+        coverageIstanbulReporter: {
+            reports: ['text-summary', 'html'],
+            fixWebpackSourcePaths: true,
+            skipFilesWithNoCoverage: false,
+            thresholds: {
+                emitWarning: false,
+                global: {
+                    statements: 80,
+                    lines: 80,
+                    branches: 80,
+                    functions: 80
```

Comment:
Thoughts on enabling code coverage requirement.

We could do it really easily by setting the `"codeCoverage"` flag for the `"test"` command in `angular.json`. However, code coverage is currently... abysmal right now

```
=============================== Coverage summary ===============================
Statements   : 63.89% ( 269/421 )
Branches     : 57.72% ( 157/272 )
Functions    : 51% ( 51/100 )
Lines        : 63.19% ( 261/413 )
```

It might be worth enabling code coverage and adjusting all of these minimum coverage thresholds to 50% or 60%, that way we don't drop any lower. But there's definitely room for improvement.